### ISSUE TITLE
Laggy and poor cpu performance after install

### ISSUE THREAD

**Issue by Star3161** [offensive-security/kali-nethunter#651]
### Device:
S7 Exynos (G930FD)

### OS version (KitKat/Lollipop/Marshmallow/Nougat):
Marshmallow

### Built from repo (date and build command) or downloaded from website (links):
kernel-nethunter-herolte-marshmallow-3.15.2-20161019-0333
kernel-nethunter-herolte-marshmallow-rebase-s7-b1

## Issue:
I noticed after installing nethunter the phone was unusually laggy so I ran some Geekbench tests and I recieved extremely poor scores whereas on the stock ROM (FXXU1BPI) the scores are normal. I know this is far from the normal scores but I believe that the kernel is somewhat limiting the CPU. 
Here are the logcats, dmesg and Geekbench screenshot: https://drive.google.com/open?id=0B9VfwWEOHA1cWUpUN3hhMFNlU3M

Thank you and I hope this gets resolved.

**Comment by jcadduono** [comment]
yeah this is an issue of framework mismatch between kernel and android, the herolte nethunter kernel is based on the note 7 sources at the moment, I suggest trying one of the various roms on xda based on Note 7 framework until I can figure this out. The lag also seems to happen on S7 kernel sources, so it seems whatever Samsung did in Note 7 framework fixed it.

**Comment by Star3161** [comment]
Using this ROM: http://forum.xda-developers.com/galaxy-s7/development/rom-epsilon-project-rom-v1-0-g930f-bpif-t3470400 seems to have fixed the lag problem. So I believe that just the stock rom is the issue.

**Comment by friiman** [comment]
Had þhe same problem. Now I installed the Superstock rom and no lag problem. But I don't have firmware to my tplink USB dongle in nethunter. Tried to install kernel from build.nethunter.com and it went laggy  again so I had to redo all again. Anyone know how to set up WiFi driver and get it to work? Thanks

**Comment by friiman** [comment]
I got samsung S7. Forgot to mention that:)

**Comment by jcadduono** [comment]
hmm, so if you install superstock rom, then install the latest kernel nightly, it goes laggy again? then superstock is not the solution

**Comment by friiman** [comment]
Yes that's what I thought to. Is there any solution to do. I might be doing something wrong. I installed the latest released kernel date.

**Comment by friiman** [comment]
Should I wipe anything after I install the kernel or just reboot?

**Comment by jcadduono** [comment]
just reboot, i guess @Star3161 is just using kernel that came with the rom? :(

**Comment by friiman** [comment]
I give it a go and try finding a way to get atheros firmware to kick in. Otherwise I wait for more knowledge 😀 thanks

**Comment by Star3161** [comment]
@jcadduono @friiman I apologize for any mis-leading information regarding my solution. I did not notice that I had not installed the kernel yet. After I had installed the kernel the lag is still there. I have also tried several Note 7 ports (http://forum.xda-developers.com/galaxy-s7/development/fastupdates-ontheedgelite-ux-v1-t3401422 & http://forum.xda-developers.com/galaxy-s7/development/rom-hydra-rom-v1-2-aroma-t3455641) as @jcadduono have advised me, but it is still lagging. I was using the latest kernel (kernel-nethunter-herolte-marshmallow-3.15.2-20161110-0008). 
I do hope there is a solution soon!

**Comment by Star3161** [comment]
Any update on this issue?

**Comment by friiman** [comment]
Hi  no  not yet. Been trying some solution but no luck. I'm new to android, I've been an iPhone guy for 7 years :) but I will keep trying. Any progress from you?

**Comment by mmollet** [comment]
Hi, iam waiting also, tried different nethunter kernels on my hero2lte but noone was able to deliver nearly the same performance as the stock kernel. Iam starting a bounty of 50$ for getting this solved in the next few days. Also it would be nice if someone could upload a stock kernel of G935FXXU1BPJG so a full restore is not always needed.

**Comment by jcadduono** [comment]
have you tried installing it on a Note 7 port ex. http://forum.xda-developers.com/s7-edge/development/rom-n7-rom-port-romcontrol-t3430358

**Comment by mmollet** [comment]
does this port has the same functions (drivers) as the nethunter kernel ? if iam not able to use the nethunter apps, how does it helps ? or, are you just interessted if this would be a possible source for building a new nethunter kernel ?

**Comment by friiman** [comment]
That sounds good, I will also pay 50 $ for a laggy free kernel. I think I factory restored  my phone over 10 times now in 3 weeks :)

**Comment by smpwojcik** [comment]
I'm really interested. I found flashing the Hydra rom ported from note 7 did work, but I lost features I enjoyed in the stock rom (Volte, video calls, only having 4G, even though people day its the same thing. Just doesn't feel right though.) https://github.com/offensive-security/kali-nethunter/wiki#30-building-nethunter says the phones are supported, but I am unable to find any information on how to build it myself, so I would spend the night learning just for that. Hahah.
